### ISSUE TITLE
macOS 10.15.1: Failed to build hie (error while building package shake-0.18.3)

### ISSUE THREAD

**Issue by awave1** [haskell/haskell-ide-engine#1574]
Hi,

I'm trying to build haskell-ide-engine from source on macOS 10.15.1, but the build keeps failing.
Here's the output:

```sh
$ stack ./install.hs hie-8.6.5

--  While building package shake-0.18.3 using:
      /Users/awave/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-osx/Cabal-2.4.0.1 build --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
Progress 1/2
```

Now, I tried running `/Users/awave/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-osx/Cabal-2.4.0.1 build --ghc-options " -fdiagnostics-color=always"` directly and it gives me the following:

```
Run the 'configure' command first.
```

I'm not entirely sure what I need to 'configure'. I've tried reinstalling `stack` as well, but that did not solve the issue.

Any help is appreciated.
Thanks!

**Comment by jneira** [comment]
Hi! well it works in most cases, not sure what would be the cause but we can try to:
* delete .stack-work inside the install directory: `stack clean --full --stack-yaml install\shake.yaml`
* run the configure phase (i.e. prepare for building) with `stack build --only-configure --stack-yaml install\shake.yaml` before trying to run the `install.hs` script

**Comment by awave1** [comment]
@jneira, hmm I tried running the config phase (after I cleaned `.stack-work`, as suggested), but that didn't finish successfully either:

```
shake      > [71 of 71] Compiling Run
...
shake      > ld: symbol(s) not found for architecture x86_64
shake      > clang: error: linker command failed with exit code 1 (use -v to see invocation)
shake      > `gcc' failed in phase `Linker'. (Exit code: 1)

--  While building package shake-0.18.3 using:
      ~/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-osx/Cabal-2.4.0.1 build --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
Progress 1/2
```

Full output: https://gist.github.com/awave1/cc63e5e8815bbe796fa256533e6fa5b6

I assume something might be wrong with `gcc` on my machine?

**Comment by jneira** [comment]
mmm, doesn't it list the concrete symbols not found? There are several issues about link errors involving haskell and macos.
btw, a simple `stack build` in hie project root works?

**Comment by awave1** [comment]
I tried doing those simple steps, it still didn't work. So the problem wasn't with hie, fortunately.

I had [`gnu-binutils`](https://formulae.brew.sh/formula/binutils) installed on my machine and I found that someone had similar issue: https://github.com/commercialhaskell/stack/issues/4830#issuecomment-492812769.

After uninstalling those, I got hie to build successfully 😅 (and `stack` works normally too)
Closing this issue, and thanks for the help 🙂
